pkg/virt-handler/mount-manager/hotplug-disk/mount_test.go

```diff
@@ -869,52 +715,36 @@ var _ = Describe("HotplugVolume", func() {
 				return []byte("Success"), nil
 			}
 
-			expectedPaths := []string{targetFilePath, blockVolume}
 			capturedPaths := []string{}
 
 			ownershipManager.EXPECT().SetFileOwnership(gomock.Any()).Times(2).DoAndReturn(func(path *safepath.Path) error {
 				capturedPaths = append(capturedPaths, unsafepath.UnsafeAbsolute(path.Raw()))
 				return nil
 			})
 
-			err = m.Mount(vmi)
-			Expect(err).ToNot(HaveOccurred())
-			By("Verifying there are 2 records in tempDir/1234")
-			record := &vmiMountTargetRecord{
-				MountTargetEntries: []vmiMountTargetEntry{
-					{
-						TargetFile: targetFilePath,
-					},
-					{
-						TargetFile: blockVolume,
-					},
+			record := []mountutils.MountTargetEntry{
+				{
+					TargetFile: targetFilePath,
+				},
+				{
+					TargetFile: blockVolume,
 				},
-				UsesSafePaths: true,
 			}
-			expectedBytes, err := json.Marshal(record)
-			Expect(err).ToNot(HaveOccurred())
-			bytes, err := os.ReadFile(filepath.Join(tempDir, string(vmi.UID)))
-			Expect(err).ToNot(HaveOccurred())
-			Expect(bytes).To(Equal(expectedBytes))
-			_, err = os.Stat(targetFilePath)
-			Expect(err).ToNot(HaveOccurred())
-			_, err = os.Stat(blockVolume)
+			mockRecorder.EXPECT().GetMountRecord(gomock.Any()).Return(record, nil)
+			mockRecorder.EXPECT().SetMountRecord(gomock.Any(), gomock.Any()).Return(nil).Times(2)
+			err = m.Mount(vmi)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(capturedPaths).To(ContainElements(expectedPaths))
-
 			volumeStatuses = make([]v1.VolumeStatus, 0)
 			volumeStatuses = append(volumeStatuses, v1.VolumeStatus{
 				Name: "permanent",
 			})
 			vmi.Status.VolumeStatus = volumeStatuses
+
+			mockRecorder.EXPECT().GetMountRecord(gomock.Any()).Return(record, nil)
+			mockRecorder.EXPECT().DeleteMountRecord(gomock.Any()).Return(nil)
 			err = m.Unmount(vmi)
+
 			Expect(err).ToNot(HaveOccurred())
-			_, err = os.ReadFile(filepath.Join(tempDir, string(vmi.UID)))
```

Comment:
This can stay or not?

Reply:
I think it is redundant, I have added the tests that check the file creation and deletion in the recorder package. In this way, we can achieve a better separation between the recorder logic and the part related to hotplug